src/indexes/text/fuzzy.h

```diff
@@ -1,43 +1,202 @@
+    return results;
+  }
 
-//
-// Takes a root word and iterates over the words that are a fuzzy match
-//
-struct FuzzyWordIterator : public WordIterator {
-  FuzzyWordIterator(indexes::Text& index, absl::string_view word,
-                    size_t distance);
-  virtual bool Done() const override;
-  virtual void Next() override;
-  virtual absl::string_view GetWord() const override;
-  virtual std::unique_ptr<WordIterator> Clone() const override;
-
-  absl::string_view operator*() const { return GetWord(); }
-};
 
+ private:
+static void SearchRecursive(
+    typename RadixTree<Target>::PathIterator iter,
+    absl::string_view pattern, 
+    size_t max_distance,
+    std::string word,  // Current word being built 
+    char prev_char,    // Previous character (for transposition detection)
+    std::vector<size_t>& prev_prev,  // Row i-2 of DP matrix (for transposition)
+    std::vector<size_t>& prev,       // Row i-1 of DP matrix (previous row)
+    std::vector<size_t>& curr,       // Row i of DP matrix (current row being computed)
+    std::vector<std::pair<std::string, Target>>& results) {
+  // ITERATE OVER SIBLINGS at current tree level
+  while (!iter.Done()) {
+    absl::string_view path = iter.GetPath();
+    std::cout << "DEBUG:   path='" << path << "' iter.IsWord()=" << iter.IsWord() << " iter.CanDescend()=" << iter.CanDescend() << std::endl;
+    std::string new_word = word;
+
+    // SAVE STATE: Each sibling must start with same parent state    
+    auto saved_prev_prev = prev_prev;
+    auto saved_prev = prev;
+    char saved_prev_char = prev_char;
+
+    // Process each character in the path
+    for (char ch : path) {
+      new_word += ch;
+      char lower_ch = std::tolower(static_cast<unsigned char>(ch));
+      
+      curr[0] = new_word.length();
+      size_t min_dist = curr[0];
+      
+      for (size_t i = 1; i <= pattern.length(); ++i) {
+        char lower_pat = std::tolower(
+            static_cast<unsigned char>(pattern[i - 1]));
+        size_t cost = (lower_ch == lower_pat) ? 0 : 1;
+        
+        curr[i] = std::min({
+            prev[i] + 1,           // Deletion
+            curr[i - 1] + 1,       // Insertion
+            prev[i - 1] + cost     // Substitution
+        });
+        
+        // Damerau-Levenshtein: transposition
+        if (i > 1 && new_word.length() > 1 && 
+            lower_ch == std::tolower(
+                static_cast<unsigned char>(pattern[i - 2])) &&
+            lower_pat == std::tolower(
+                static_cast<unsigned char>(prev_char))) {
+          curr[i] = std::min(curr[i], prev_prev[i - 2] + cost);
+        }
+        
+        min_dist = std::min(min_dist, curr[i]);
+      }
+      
+      prev_prev.swap(prev);
+      prev.swap(curr);
+      prev_char = ch;
+    }
+    // NOW check if we should prune this subtree
+    size_t min_dist = *std::min_element(prev.begin(), prev.end());
+    bool should_prune = (min_dist > max_distance);
```

Comment:
Will this cause us to exit too early? At the root the path is just the empty string, which will usually have a fuzzy distance greater than the max distance